nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,543 @@
+            '';
+          };
+
+          staticRoot = mkOption {
+            type = types.str;
+            default = "/srv/funkwhale/static";
+            description = ''
+              Where static files (such as API css or icons) should be compiled on your system ? Ensure this directory actually exists.
+            '';
+          };
+
+          djangoSecretKey = mkOption {
+            type = types.str;
+            description = ''
+              Django secret key. Generate one using `openssl rand -base64 45` for example.
+            '';
+            example = "6VhAWVKlqu/dJSdz6TVgEJn/cbbAidwsFvg9ddOwuPRssEs0OtzAhJxLcLVC";
+          };
+        };
+
+        musicDirectoryPath = mkOption {
+          type = types.str;
+          default = "/srv/funkwhale/music";
+          description = ''
+            In-place import settings.
+          '';
+        };
+
+        enableRaven = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Sentry/Raven error reporting (server side).
+            Enable Raven if you want to help improve funkwhale by
+            automatically sending error reports to the funkwhale developers Sentry instance.
+            This will help them detect and correct bugs.
+          '';
+        };
+
+        ravenDsn = mkOption {
+          type = types.str;
+          default = "https://44332e9fdd3d42879c7d35bf8562c6a4:0062dc16a22b41679cd5765e5342f716@sentry.eliotberriot.com/5";
+          description = ''
+            Sentry/Raven DSN.
+            The default is the Funkwhale developers instance DSN.
+          '';
+        };
+
+      };
+    };
+
+    config = mkIf cfg.enable {
+      assertions = [
+        { assertion = cfg.database.passwordFile != null || cfg.database.password != "" || cfg.database.socket != null;
+          message = "one of services.funkwhale.database.socket, services.funkwhale.database.passwordFile, or services.funkwhale.database.password must be set";
+        }
+        { assertion = cfg.database.createLocally -> cfg.database.user == cfg.user;
+          message = "services.funkwhale.database.user must be set to ${cfg.user} if services.funkwhale.database.createLocally is set true";
+        }
+        { assertion = cfg.database.createLocally -> cfg.database.socket != null;
+          message = "services.funkwhale.database.socket must be set if services.funkwhale.database.createLocally is set to true";
+        }
+        { assertion = cfg.database.createLocally -> cfg.database.host == "localhost";
+          message = "services.funkwhale.database.host must be set to localhost if services.funkwhale.database.createLocally is set to true";
+        }
+      ];
+
+      users.users = optionalAttrs (cfg.user == "funkwhale") (singleton
+        { name = "funkwhale";
+          group = cfg.group;
+        });
+
+      users.groups = optionalAttrs (cfg.group == "funkwhale") (singleton { name = "funkwhale"; });
```

Comment:
Do `users.groups.funkwhale = mkIf (cfg.group == "funkwhale") {}` instead, and similarly for `users.users`